dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java

```diff
@@ -141,44 +185,44 @@ public PageView getPageMetadataRendered(final HttpServletRequest request, final
         return getPageMetadata(request, response, user, uri, true, PageMode.get(request));
     }
 
+    public String getPageRendered(final HttpServletRequest request, final HttpServletResponse response, final User user,
+                                  final String uri, final PageMode mode) throws Exception {
+
+        final HTMLPageAsset page =  this.getPage(request, user, uri, mode);
+        return this.getPageRendered(page, request, response, user, mode);
+    }
+
     @CloseDB
-    public String getPageRendered(final HttpServletRequest request, final
-    HttpServletResponse response, final User user, final String uri, PageMode mode) throws DotDataException, DotSecurityException, IOException {
-        
+    public String getPageRendered(final HTMLPageAsset page, final HttpServletRequest request,
+                                  final HttpServletResponse response, final User user, final PageMode mode) throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)